src/pfe/file-watcher/server/src/utils/kubeutil.ts

```diff
@@ -333,6 +333,98 @@ export async function installChart(projectID: string, deploymentName: string, ch
     return response;
 }
 
+export async function exposeOverIngress(projectID: string, projectName: string, ingressDomain: string, isHTTPS: boolean): Promise<string> {
+    let ownerReferenceName: string;
+    let ownerReferenceUID: string;
+    let serviceName: string;
+    let servicePort: number;
+
+    try {
+        let resp: any = undefined;
+
+         // Get the deployment name and uid labaled with the unique project ID
+        resp = await k8sClient.apis.apps.v1.namespaces(KUBE_NAMESPACE).deployments.get({ qs: { labelSelector: "projectID=" + projectID } });
+        for ( let i = 0 ; i < resp.body.items.length ; i++ ) {
+            ownerReferenceName = resp.body.items[i].metadata.name;
+            ownerReferenceUID = resp.body.items[i].metadata.uid;
+        }
+
+        // Get the service name and port labeled with the unique project ID
+        resp = await k8sClient.api.v1.namespaces(KUBE_NAMESPACE).services.get({ qs: { labelSelector: "projectID=" + projectID } });
+        for ( let i = 0 ; i < resp.body.items.length ; i++ ) {
+            serviceName = resp.body.items[i].metadata.name;
+            servicePort = parseInt(resp.body.items[i].spec.ports[0].port, 10);
+        }
+
+    } catch (err) {
+        logger.logProjectError("Unable to retrieve project deployment or service", projectID);
+        throw err;
+    }
+
+    try {
+
+        // Create and deploy the ingress on Kube if one for the project doesn't already exist
+        const resp = await k8sClient.apis.extensions.v1beta1.namespaces(KUBE_NAMESPACE).ingresses.get({ qs: { labelSelector: "projectID=" + projectID } });
+        if (resp.body.items.length == 0) {
```

Comment:
Now that I think about it, I think we'll need to refresh the ingress each time, in case the port does change.

This should be a relatively quick change.